pkg/reconciler/queue_pipelineruns.go

```diff
@@ -34,6 +34,12 @@ func (r *Reconciler) queuePipelineRun(ctx context.Context, logger *zap.SugaredLo
 		return fmt.Errorf("updateError: %w", err)
 	}
 
+	// merge local repo with global repo here in order to derive settings from global
+	// for further concurrency and other operations.
+	if r.globalRepo, err = r.repoLister.Repositories(r.run.Info.Kube.Namespace).Get(r.run.Info.Controller.GlobalRepository); err == nil && r.globalRepo != nil {
```

Comment:
do we need to do listing here? global repo is available in not available in reconciler object?

Reply:
@piyush-garg I confirmed that we need to fetch it, global repo wasn't there before `queue_pipelineruns`.

Reply:
ack, got it